worker/src/main/java/gov/cms/ab2d/worker/bfdhealthcheck/BFDHealthCheck.java

```diff
@@ -0,0 +1,64 @@
+package gov.cms.ab2d.worker.bfdhealthcheck;
+
+import gov.cms.ab2d.bfd.client.BFDClient;
+import gov.cms.ab2d.common.dto.PropertiesDTO;
+import gov.cms.ab2d.common.service.PropertiesService;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.dstu3.model.CapabilityStatement;
+import org.hl7.fhir.dstu3.model.Enumerations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static gov.cms.ab2d.common.util.Constants.MAINTENANCE_MODE;
+
+@Component
+@Slf4j
+class BFDHealthCheck {
+
+    @Autowired
+    private PropertiesService propertiesService;
+
+    @Autowired
+    private BFDClient bfdClient;
+
+    @Value("${bfd.health.check.consecutive.successes}")
+    private int consecutiveSuccessesToBringUp;
+
+    @Value("${bfd.health.check.consecutive.failures}")
+    private int consecutiveFailuresToTakeDown;
+
+    // Nothing else should be calling this component except for the scheduler, so keep
+    // state here
+    private int consecutiveSuccesses = 0;
+
+    private int consecutiveFailures = 0;
+
```

Comment:
To address the issues mentioned below, I think we need another instance variable such as `private Status bfdStatus` where `Status` would be a private `enum` with `UP` and `DOWN` values. This variable would hold the currently known status of BFD.

The code below would count failures _only_ when the status is UP and vice versa.